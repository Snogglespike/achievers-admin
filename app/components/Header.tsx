import { Link } from "@remix-run/react";

export default function Header() {
  return (
    <nav className="flex flex-wrap items-center justify-between bg-blue-500 p-2">
      <div className="mr-6 flex flex-shrink-0 items-center text-white">
        <img className="mr-4 w-16 rounded" src="/images/logo.png" alt="Logo" />
        <span className="text-xl font-semibold tracking-tight">
          Achievers WA
        </span>
      </div>
      <div className="block lg:hidden">
        <button className="flex items-center rounded border border-white px-3 py-2 text-white">
          <svg
            className="h-3 w-3 fill-current"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>Menu</title>
            <path d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z" />
          </svg>
        </button>
      </div>
      <div className="block w-full flex-grow lg:flex lg:w-auto lg:items-center">
        <div className="lg:flex-grow">
          <Link
            to="/users"
            className="mt-4 mr-6 block text-white hover:text-white lg:mt-0 lg:inline-block"
          >
            Users
          </Link>
          <Link
            to="/chapters"
            className="mt-4 block text-white hover:text-white lg:mt-0 lg:inline-block"
          >
            Chapters
          </Link>
        </div>
        <div>
          <Link
            to="/logout"
            className="mt-4 inline-block rounded border border-white px-4 py-2 text-sm leading-none text-white hover:border-transparent hover:bg-white hover:text-teal-500 lg:mt-0"
          >
            Logout
          </Link>
        </div>
      </div>
    </nav>
  );
}
