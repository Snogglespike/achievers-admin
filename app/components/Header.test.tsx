import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { MemoryRouter } from "react-router";

import Header from "~/components/Header";

test("Header snapshot", async () => {
  const { baseElement } = render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );

  expect(baseElement).toMatchSnapshot();
});
