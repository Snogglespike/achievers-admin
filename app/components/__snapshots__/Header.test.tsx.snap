// Vitest Snapshot v1

exports[`Header snapshot 1`] = `
<body>
  <div>
    <nav
      class="relative flex h-14 shrink-0 flex-wrap items-center justify-between bg-blue-500"
    >
      <a
        class="mr-6 ml-2 flex flex-shrink-0 items-center text-white"
        href="/users"
      >
        <img
          alt="Logo"
          class="mr-4 w-16 rounded"
          src="/images/logo.png"
        />
        <span
          class="text-xl font-semibold tracking-tight"
        >
          Achievers WA
        </span>
      </a>
      <div
        class="mr-2 block lg:hidden"
      >
        <button
          class="flex items-center rounded border border-white px-3 py-2 text-white"
        >
          <svg
            class="h-3 w-3 fill-current"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Menu
            </title>
            <path
              d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"
            />
          </svg>
        </button>
      </div>
      <div
        class="absolute top-14 z-10 w-full flex-grow bg-blue-500 p-2 lg:static lg:flex lg:w-auto lg:items-center hidden"
      >
        <div
          class="lg:flex-grow"
        >
          <a
            class="mt-4 mr-6 block text-white hover:text-white lg:mt-0 lg:inline-block"
            href="/users"
          >
            Users
          </a>
          <a
            class="mt-4 block text-white hover:text-white lg:mt-0 lg:inline-block"
            href="/chapters"
          >
            Chapters
          </a>
        </div>
        <div>
          <a
            class="mt-4 inline-block rounded border border-white px-4 py-2 text-sm leading-none text-white hover:border-transparent hover:bg-white hover:text-teal-500 lg:mt-0"
            href="/logout"
          >
            Logout
          </a>
        </div>
      </div>
    </nav>
  </div>
</body>
`;
